### ISSUE TITLE
[Moderate] Tabs: Tab widget is missing appropriate roles and/or attributes

### ISSUE THREAD

**Issue by ntallman** [APTrust/dart#589]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87705 |
| **Checkpoint** | 4.1 Compatible › Custom Controls (4.1.2.b) |
| **Impact** | Moderate |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 01 Settings Workflow |
| **URL** | <https://aptrust.org/documentation-page/dart-digital-archivists-resource-tool/> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Date Reported** | 05/11/26 at 9:02 am |
| **Issue Status** | Open |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

The element appears and functions like a tab/tabpanel interface but is missing required ARIA role(s) and/or attribute(s).
In Settings menu, select BagIt Profiles, then New, then accept the default of AP Trust, then select Create.  The tabs (About, Info, General, etc.) do not provide the expected behavior of tab in and use arrow keys to select another.  Additionally, the last tab is an expandable menu, which adds additional complexity. See Settings User Experience video at 6:25.

## Recommendation

Rule : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

How To Fix: 
Tabs are a set of layered sections of content, known as tab panels, that display one panel of content at a time. Each tab panel has an associated tab element, that when activated, displays the panel. The list of tab elements is arranged along one edge of the currently displayed panel, most commonly the top edge.

Fix this issue by using the following roles and attributes:
1. The element that serves as the container for the set of tabs has role tablist.
2. Each element that serves as a tab has role tab and is contained within the element with role tablist.
3. Each element that contains the content panel for a tab has role tabpanel.
4. Each element with role tab has the property aria-controls referring to its associated tabpanel element.
5. The active tab element has the state aria-selected set to true and all other tab elements have it set to false.
6. Each element with role tabpanel has the property aria-labelledby referring to its associated tab element.
7. If the tablist element is vertically oriented, it has the property aria-orientation set to vertical. The default value of aria-orientation for a tablist element is horizontal.

View the full ARIA design pattern including expected keyboard interaction and examples: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

Reference: 
WAI-ARIA Authoring Practices: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

Background: 
Every user interface control must have a role along with any applicable states and properties so that screen reader users know how to interact with the control. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, and their necessary states and properties - such as the checked/unchecked state of a checkbox - are automatically conveyed so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) and any applicable states and properties using ARIA as well as expected keyboard interactions.

## Screenshots

![01_tab](https://raw.githubusercontent.com/APTrust/dart/master/docs/accessibility-audit/screenshots/87705/01_tab.png)
